### PR TITLE
test: hard code date values in Date_column_types_validation_spec

### LINKED ISSUE #36756: [Fix]: Flaky Date_column_types_validation_spec
### SubTasks

The problem is in first spec, that the spec is being fed a specific date data from file(app/client/cypress/fixtures/tableDateColumnTypes.ts), while it is trying to work on relative dates(i.e. today and tomorrow)
Which means that when date picker opens up it opens with September 2024 in view - And we are trying to select tomorrow (8th Oct) which is not in the view.
All subsequent failures are related to that.

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
**Problem**
In the test spec, date data is fed directly from a file `(app/client/cypress/fixtures/tableDateColumnTypes.ts)`, which includes specific, static dates. However, the test is designed to handle relative dates (e.g., "today" and "tomorrow"). As a result, when the date picker opens, it displays a fixed date (e.g., September 2024), while the test is attempting to select a date like October 8 (tomorrow’s date), which is out of view.

EE PR - https://github.com/appsmithorg/appsmith-ee/pull/5312

Fixes #36756

## Automation

/ok-to-test tags="@tag.Table"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11270917513>
> Commit: 9ac8789c2c00838cef45fcc1ad712e1afc419942
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11270917513&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Table`
> Spec:
> <hr>Thu, 10 Oct 2024 09:53:10 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **Bug Fixes**
	- Enhanced clarity and specificity in end-to-end tests for date column validation in the table widget.
	- Updated assertions to directly compare against a specific hardcoded date.

- **Refactor**
	- Removed the `getFormattedTomorrowDates` method from the Table class, streamlining date formatting functionality.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_types_validation_spec.ts [hunk 1]
```diff
@@ -43,17 +43,16 @@ describe(
...
       );
 
-      // Check that date is set in column
+      // Check that the date is set in column
       table
         .ReadTableRowColumnData(row, column, "v2")
-        .then((val) =>
-          expect(val).to.equal(table.getFormattedTomorrowDates().isoFormat),
-        );
+        .then((val) => expect(val).to.equal("2024-09-26"));
```
Comment (sagar-qa007) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_types_validation_spec.ts:55]: Do we think this is also hard coded?

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_types_validation_spec.ts (hunk 1 above)
Comment (jacquesikot) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_types_validation_spec.ts:55]: Yes, this is a hardcoded date value, i.e when the fixed date from the Table data is used, this hardcoded date here is always visible in the date picker viewport and will be selected.

**Review by rahulbarwal — APPROVED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**